gcloud-java-resourcemanager/src/main/java/com/google/gcloud/resourcemanager/testing/LocalResourceManagerHelper.java

```diff
@@ -0,0 +1,593 @@
+          return "Project " + project.getProjectId() + " has an invalid name."
+              + " See https://cloud.google.com/resource-manager/reference/rest/" + VERSION
+              + "/projects for more information.";
+        }
+      }
+    }
+    if (project.getLabels() != null) {
+      if (project.getLabels().size() > 256) {
+        return "Project " + project.getProjectId() + " exceeds the limit of 256 labels.";
+      }
+      for (Map.Entry<String, String> entry : project.getLabels().entrySet()) {
+        if (!isValidIdOrLabel(entry.getKey(), 1, 63)
+            || !isValidIdOrLabel(entry.getValue(), 0, 63)) {
+          return "Project " + project.getProjectId() + " has an invalid label entry."
+              + " See https://cloud.google.com/resource-manager/reference/rest/" + VERSION
+              + "/projects for more information.";
+        }
+      }
+    }
+    return null;
+  }
+
+  private static final boolean isValidIdOrLabel(String value, int minLength, int maxLength) {
+    for (char c : value.toCharArray()) {
+      if (c != '-' && !Character.isDigit(c)
+          && (!Character.isLetter(c) || !Character.isLowerCase(c))) {
+        return false;
+      }
+    }
+    if (value.length() > 0 && (!Character.isLetter(value.charAt(0)) || value.endsWith("-"))) {
+      return false;
+    }
+    return value.length() >= minLength && value.length() <= maxLength;
+  }
+
+  Response create(Project project) {
+    project.setLifecycleState("ACTIVE");
+    project.setProjectNumber(Math.abs(PROJECT_NUMBER_GENERATOR.nextLong() % Long.MAX_VALUE));
+    project.setCreateTime(ISODateTimeFormat.dateTime().print(System.currentTimeMillis()));
+    Response response;
+    String customErrorMessage = checkForProjectErrors(project);
+    if (customErrorMessage != null) {
+      response = Error.INVALID_ARGUMENT.response(customErrorMessage);
+    } else if (projects.containsKey(project.getProjectId())) {
+      response = Error.ALREADY_EXISTS.response(
+          "A project with the same project ID (" + project.getProjectId() + ") already exists.");
+    } else {
+      projects.put(project.getProjectId(), project);
+      try {
+        String createdProjectStr = jsonFactory.toString(project);
+        response = new Response(HttpURLConnection.HTTP_OK, createdProjectStr);
+      } catch (IOException e) {
+        response =
+            Error.INTERNAL_ERROR.response("Error serializing project " + project.getProjectId());
+      }
+    }
+    return response;
+  }
+
+  Response delete(String projectId) {
+    Project project = projects.get(checkNotNull(projectId));
+    Response response;
+    if (project == null) {
+      // when possible, change this to 404 (#440)
+      response = Error.PERMISSION_DENIED.response(
+          "Error when deleting " + projectId + " because the project was not found.");
+    } else if (!project.getLifecycleState().equals("ACTIVE")) {
```

Comment:
Made an issue to deal with synchronization because I want to think a little bit more about any potential deadlocks (since there's already a lock on the concurrentHashMap)